### ISSUE TITLE
Inconsistent result from a torch model and a torchscript model

### ISSUE THREAD

**Issue by srib** [pytorch/pytorch#56060]
## 🐛 Bug

I am trying to compare the results from a `torchvision` model and a model that has been converted to `torchscript` and I get inconsistent results.

## To Reproduce

Environment: I am currently using `torch` [image](https://hub.docker.com/layers/pytorch/pytorch/1.8.0-cuda11.1-cudnn8-devel/images/sha256-f915e85ca58c8d6c072de080288dd66965ca623744a8392cf5ea0c2df485b23c?context=explore) from DockerHub.

```python
import torch
import torchvision

# Load a segmentation model from torchvision
model = torchvision.models.segmentation.fcn_resnet101(pretrained=True, progress=True, num_classes=21)

example = torch.ones(1, 3, 224, 224) 

out = model(example)['out']

print(type(out), out.shape)

traced_script_module = torch.jit.script(model)

output = traced_script_module(example)['out']

print(type(output), output.shape)

print(out[0, 1, :, :], output[0, 1, :, :])
```
Run as follows:
```bash
docker run --rm -it --init --gpus=all --net=host --ipc=host -v "$PWD":/usr/src/myapp torchscript_test /opt/conda/bin/python /usr/src/myapp/torchscript_converter.py
```

The output tensors are significantly different.

## Expected behavior

`out` and `output` in the case above should be identical.

## Environment

I am using the docker image from [here](https://hub.docker.com/layers/pytorch/pytorch/1.8.0-cuda11.1-cudnn8-devel/images/sha256-f915e85ca58c8d6c072de080288dd66965ca623744a8392cf5ea0c2df485b23c?context=explore)
## Additional context

cc @gmagogsfm

**Comment by gmagogsfm** [comment]
Hi @srib,

It seems the unscripted model produces different result when called multiple times on same input. So I don't think this behavior is related to TorchScript. My best guess so far is that the model contains randomness or maybe it is still in training mode.

**Comment by srib** [comment]
Thanks for responding, @gmagogsfm. I hadn't called `.eval()` on it.
